protocol/chainlib/grpc.go

```diff
@@ -341,6 +341,10 @@ func (cp *GrpcChainProxy) SendNodeMsg(ctx context.Context, ch chan interface{},
 	response := msgFactory.NewMessage(methodDescriptor.GetOutputType())
 	err = conn.Invoke(connectCtx, "/"+nodeMessage.Path, msg, response)
 	if err != nil {
+		if strings.Contains(err.Error(), context.DeadlineExceeded.Error()) {
```

Comment:
string contains is not the way to check if there was a context deadline you have connectCtx.Error():
ctx.Err() == context.DeadlineExceeded